src/BugsnagUnity/FileManager.cs

```diff
@@ -106,30 +128,64 @@ internal static void CacheSession(string reportId)
             }
         }
 
+        internal static void CacheEvent(string reportId)
+        {
+            var eventReport = GetPendingEventReport(reportId);
+            if (eventReport == null)
+            {
+                return;
+            }
+            using (var stream = new MemoryStream())
+            using (var reader = new StreamReader(stream))
+            using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { AutoFlush = false })
+            {
+                SimpleJson.SerializeObject(eventReport.GetSerialisableEventReport(), writer);
+                writer.Flush();
+                stream.Position = 0;
+                var jsonString = reader.ReadToEnd();
+                var path = EventsDirectory + "/" + eventReport.Id + ".event";
+                WriteToDisk(jsonString, path);
+                CheckForMaxCachedEvents();
+            }
+        }
+
+        private static void CheckForMaxCachedEvents()
+        {
+            var filesCount = _cachedEvents.Length;
+            while (filesCount > _configuration.MaxPersistedEvents)
+            {
+                RemoveOldestFile(_cachedEvents);
+                filesCount = _cachedEvents.Length;
+            }
+        }
+
         private static void CheckForMaxCachedSessions()
         {
             var filesCount = _cachedSessions.Length;
             while(filesCount > _configuration.MaxPersistedSessions)
             {
-                RemoveOldestSession();
+                RemoveOldestFile(_cachedSessions);
                 filesCount = _cachedSessions.Length;
             }
         }
 
-        private static void RemoveOldestSession()
+        private static void RemoveOldestFile(string[] filePaths)
         {
-            var oldestSession = string.Empty;
+            var oldestFilePath = string.Empty;
             var oldestMillis = default(double);
-            foreach (var cachedSessionPath in _cachedSessions)
+            foreach (var filePath in filePaths)
             {
-                var milliesSinceCreated = (DateTime.UtcNow - File.GetCreationTimeUtc(cachedSessionPath)).TotalMilliseconds;
+                var milliesSinceCreated = (DateTime.UtcNow - File.GetCreationTimeUtc(filePath)).TotalMilliseconds;
```

Comment:
Again unrelated to this change but if we're looking for the oldest creation time why don't we just compare the datetime rather than doing a calculation on it first?

E.g. `if (File.GetCreationTimeUtc(filePath)) < oldestFileDateTime)` (and initialise `oldestFileDateTime` to `DateTime.UtcNow` (or `DateTime.Max`).

Alternatively we could use `OrderBy()` and then remove the first / last.  If you ordered by at the time you populated the field you wouldn't need to keep searching for the oldest one as you'd know.

Reply:
Rewrote the method with OrderBy 👍